boards/v2/gigadevice/index.rst

```diff
@@ -0,0 +1,10 @@
+.. _boards-gigadevice:
+
+Gigadevice Boards
```

Comment:
Is it time we start looking at Sphinx/python tooling to automate some of the things around boards' documentation, in particular the generation of the toctree?
 
Reason I'm thinking this is that here you have GigaDevice spelled Gigadevice (lowercase d), *and* the boards' vendor is set as "GigaDevice Semiconductor". Looks like the kind of inconsistencies that, while certainly not the end of the world, can easily spread to many places.

If board.yml starts adopting the vendor prefix, a first step could be that we automatically add entries to `substititions.txt` so that something like `|vendor-gd|` gets replaced by the associated entry in `vendor-prefixes.txt`

Reply:
@kartben It would be nice to implement a new extension to display an interactive list of boards, where you can filter by vendor, soc, etc. It should now be easier with the new board.yml file. Regarding vendor, I think `board.yml` should contain a `vendor: $prefix`, and use a database (like we do for DT), cc: @tejlmand

Reply:
> I think board.yml should contain a vendor: $prefix, and use a database (like we do for DT), cc: @tejlmand

Not against the idea, but we need to also remember that it should be easy for oot users to just create their own board.
Of course such users can provide a custom vendor-prefixes.txt file, but then it starts to require a bit more integration in multiple places.

I think we should have an enhancement issue, and then link that issue in here: https://github.com/zephyrproject-rtos/zephyr/issues/51831

Reply:
for now: using text found in vendor-prefixes, so at least we're consistent everywhere.